sklearn/decomposition/tests/test_pca.py

```diff
@@ -389,6 +390,15 @@ def test_pca_validation():
                                     PCA(n_components, svd_solver=solver)
                                     .fit, data)
 
+        n_components = 1.0
+        type_ncom = type(n_components)
+        assert_raise_message(ValueError,
+                             "n_components={} must be of type int "
+                             "when greater or equal to 1, was of type={}"
+                             .format(n_components, type_ncom),
+                             PCA(n_components).fit, data)
```

Comment:
(1)Sems that you are not taking into account different svd_solvers. Remember that you are in the loop `for solver in solver_list:`, so it might be better to have something like `PCA(n_components=..., svd_solver=solver)`.
(2)Seems that you are using `data` out of the loop which defined `data`, it might be better to use X ot X.T, here, maybe X.T(n_samples > n_features) is better.

Reply:
Ye, that was my intention, but the morning coffee hadn't kicked in. Will fix